tools/cjxl_ng_main.cc

```diff
@@ -381,50 +614,43 @@ jxl::Status GetPixeldata(const jxl::PaddedBytes& image_data,
   return true;
 }
 
-void set_usage_message_and_version(const char* argv0) {
-  gflags::SetUsageMessage(
-      "JPEG XL-encodes an image.\n"
-      " Input format can be one of: "
-#if JPEGXL_ENABLE_APNG
-      "PNG, APNG, "
-#endif
-#if JPEGXL_ENABLE_GIF
-      "GIF, "
-#endif
-#if JPEGXL_ENABLE_JPEG
-      "JPEG, "
-#endif
-      "PPM, PFM, PGX.\n  Sample usage:\n" +
-      std::string(argv0) + " <source_image_filename> <target_image_filename>");
-  uint32_t version = JxlEncoderVersion();
-
-  gflags::SetVersionString(std::to_string(version / 1000000) + "." +
-                           std::to_string((version / 1000) % 1000) + "." +
-                           std::to_string(version % 1000));
-}
-
 }  // namespace
 
 int main(int argc, char** argv) {
-  std::cerr << "Warning: This is work in progress, consider using cjxl instead!"
-            << std::endl;
-  set_usage_message_and_version(argv[0]);
-  // TODO(firsching): rethink --help handling
-  gflags::ParseCommandLineNonHelpFlags(&argc, &argv, /*remove_flags=*/true);
-  if (FLAGS_help) {
-    FLAGS_help = false;
-    FLAGS_helpshort = true;
+  std::string version = jpegxl::tools::CodecConfigString(JxlEncoderVersion());
+  jpegxl::tools::CompressArgs args;
+  jpegxl::tools::CommandLineParser cmdline;
+  args.AddCommandLineOptions(&cmdline);
+
+  if (!cmdline.Parse(argc, const_cast<const char**>(argv))) {
+    // Parse already printed the actual error cause.
+    fprintf(stderr, "Use '%s -h' for more information\n", argv[0]);
+    return jpegxl::tools::CjxlRetCode::ERR_PARSE;
   }
-  gflags::HandleCommandLineHelpFlags();
 
-  if (argc != 3) {
-    FLAGS_help = false;
-    FLAGS_helpshort = true;
-    gflags::HandleCommandLineHelpFlags();
-    return EXIT_FAILURE;
+  if (args.version) {
+    fprintf(stdout, "cjxl %s\n", version.c_str());
+    fprintf(stdout, "Copyright (c) the JPEG XL Project\n");
+    return jpegxl::tools::CjxlRetCode::OK;
+  }
+
+  if (!args.quiet) {
+    fprintf(stderr, "JPEG XL encoder %s\n", version.c_str());
+  }
+
+  const char* filename_in = args.file_in;
+  const char* filename_out = args.file_out;
```

Comment:
we could keep using args.file_in, args.file_out (to be consistent with other args)

Reply:
makes sense, done